File: DotnetAPI/TWservice2/Models/Feedback.cs
using System;
using System.Collections.Generic;

namespace TWservice2.Models
{
    public partial class Feedback
    {
        public int FeedbackId { get; set; }
        public int CustomerPlanId { get; set; }
        public int Uid { get; set; }
        public int VSubscriptionId { get; set; }
        public string FeedbackText { get; set; } = null!;
        public int? Rating { get; set; }
        public DateOnly FeedbackDate { get; set; }

        public virtual CustomerSubscribedPlan CustomerPlan { get; set; } = null!;
        public virtual User UidNavigation { get; set; } = null!;
        public virtual VendorSubscriptionPlan VSubscription { get; set; } = null!;
    }
}

File: Backend/Tiffinwala/src/main/java/Tiffinwala/App/Entities/Feedback.java
package Tiffinwala.App.Entities;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDate;
@Data
@Entity
@Table(name = "Feedback")
public class Feedback {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer feedbackId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "Uid", nullable = false)
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "Customer_plan_id", nullable = false)
    private CustomerSubscribedPlans customerSubscribedPlan;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "V_Subscription_id", nullable = false)
    private VendorSubscriptionPlan vendorSubscriptionPlan;

    @Column(name = "Feedback_Text", nullable = false, length = 500)
    private String feedbackText;

    @Column(nullable = false)
    private Integer rating;



    // Constructors
    public Feedback() {}

    public Feedback(User user, CustomerSubscribedPlans customerPlan, VendorSubscriptionPlan vendorSubscriptionPlan, String feedbackText, Integer rating, LocalDate feedbackDate) {
        this.user = user;
        this.customerSubscribedPlan = customerPlan;
        this.vendorSubscriptionPlan = vendorSubscriptionPlan;
        this.feedbackText = feedbackText;
        this.rating = rating;
    }

    
}
